include/mlir/Dialect/AffineOps/AffineOps.td

```diff
@@ -223,6 +223,8 @@ def AffineIfOp : Affine_Op<"if", [ImplicitAffineTerminator]> {
     IntegerSet getIntegerSet();
     void setIntegerSet(IntegerSet newSet);
 
+    void setConditional(IntegerSet set, ArrayRef<Value *> operands);
```

Comment:
The AffineIf builders will need to be updated to allow for a variadic operands list.

Reply:
I assume you mean resizable operands list? The canonicalization here always shrinks the list. So I didn't make update it - there isn't a use case in MLIR yet that could make the operands list of AffineIf grow.

Reply:
We should guard against that then, can you document the expectation here that there are less operands? There should also be an assert for this case.

Reply:
Documented this. There is already the guard in setOperands which will assert if one tries to grow the list. Do you think we should add one more here?

Reply:
Can you reword the constraint a bit? It reads a little weird to me, something like "The size of 'operands' must not exceed the current number operands for this instance, as the operands list of AffineIf is not resizable"

Reply:
Fixed - thanks.